result-async/src/main/scala/uscala/concurrent/result/AsyncResult.scala

```diff
@@ -60,7 +58,7 @@ final class AsyncResult[+A, +B](val underlying: Future[Result[A, B]]) extends Se
     attemptRunFor(f, Duration.Inf)
 }
 
-object AsyncResult {
+object AsyncResult extends AsyncResultImplicits {
```

Comment:
We just extend the implicits trait, and the right one will get used depending on what we're compiling for.